app/modules/key_management/secret_manager.py

```
@@ -1408,3 +1445,269 @@ async def delete_all_integration_keys(
+                service, customer_id, SecretManager.WORKFLOW, db, None, workflow_id
+            )
+            return WorkflowSecretResponse(
+                service=service, workflow_id=workflow_id, api_key=secret
+            )
+        except HTTPException:
+            raise  # Re-raise HTTP exceptions
+        except Exception as e:
+            logger.error(f"Error getting workflow secret: {str(e)}")
+            raise HTTPException(
+                status_code=500, detail=f"Error retrieving workflow secret: {str(e)}"
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)